neuralpp/test/quick_tests/symbolic/normalizer_test.py

```diff
@@ -120,11 +119,65 @@ def test_quantifier_normalizer():
 
     sum_ = BasicSummation(int, i, empty_context & (j < i), if_then_else(i > 5, i + j, i))
     assert normalizer.normalize(sum_, empty_context).syntactic_eq(
-                     BasicSummation(int, i, empty_context & (j < i) & (i > 5), i + j) +
-                     BasicSummation(int, i, empty_context & (j < i) & (i <= 5), i))
+        BasicSummation(int, i, empty_context & (j < i) & (5 < i), i + j) +
+        BasicSummation(int, i, empty_context & (j < i) & ~(5 < i), i))
+
+    sum1 = if_then_else(j > 5, i + j, sum_)
 
-    # Normalization of nested quantifier expression is not supported, yet.
+    # Normalization of nested quantifier expression
     sum_ = BasicSummation(int, j, empty_context & (j < 10), if_then_else(j > 5, i + j, sum_))
-    # because SymPyQuantifierExpression is not implemented.
-    with pytest.raises(NotImplementedError):
-        normalizer.normalize(sum_, empty_context)
+    assert normalizer.normalize(sum_, empty_context).syntactic_eq(
+        BasicSummation(int, j, empty_context & (10 > j) & (j > 5), i + j) +
+        BasicSummation(int, j, empty_context & (10 > j) & ~(5 < j),
+                       BasicSummation(int, i, empty_context & (j < i) & (5 < i), i + j) +
+                       BasicSummation(int, i, empty_context & (j < i) & ~(5 < i), i)))
+
+    #          f
+    #       /    \
+    #     if A   Sum
+    #     /  \    |
+    #    B    C  B * if B>4
+    #                /    \
+    #               B+C    Sum
+    #                       |
+    #                      if B<5
+    #                      / \
+    #                     C   1
+    # should be normalized to
+    #              if A
+    #           /        \
+    #      if B>4       if B>4
+    #       /    \        /   \
+    #      f      ..   ..      f
+    #     /\                  /  \
+    #    B  Sum              C   Sum
```

Comment:
Love the example you have here! A related question: if, instead of `B>4`, we have `B==4` here, what will happen to the `B` in the left subtree?

Reply:
`B` will be replaced by `4`, since [we call `ContextSimplifier.simplify()` in `_eliminate()`](https://github.com/rodrigodesalvobraz/neuralpp/pull/30/files/202e42ac2f4a8776825662798522df898b15460b#diff-54fbb81c8dfe09a753986c9a0e95c154bad9f41e90addf62fd59579ab648f1d5R83)